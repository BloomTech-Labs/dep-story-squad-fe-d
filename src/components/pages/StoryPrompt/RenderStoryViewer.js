import React, { useState, useEffect } from 'react';
import { Document, Page, pdfjs } from 'react-pdf';
import { getStory } from '../../../api/index';
import { Header } from '../../common';
import { useOktaAuth } from '@okta/okta-react';
import { Button } from 'antd';
import { SizeMe } from 'react-sizeme';

const RenderStoryViewer = props => {
  const [numPages, setNumPages] = useState(null);
  const [pageNumber, setPageNumber] = useState(1);
  const [storyPrompt, setStoryPrompt] = useState();
  const { authState } = useOktaAuth();

  pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

  useEffect(() => {
    // ========== second argument to getStory() is hardcoded for testing ==========
    getStory(authState, 11).then(res => {
      setStoryPrompt(res.URL);
      console.log(res);
    });
  }, [authState]);

  const onDocumentLoadSuccess = ({ numPages }) => {
    setNumPages(numPages);
    setPageNumber(1);
  };
  const changePage = offset => {
    setPageNumber(prevPageNumber => prevPageNumber + offset);
  };
  const previousPage = () => {
    changePage(-1);
  };
  const nextPage = () => {
    changePage(1);
  };

  return (
    <>
      <Header backButton={true} />
      <div class="viewer-container">
        <SizeMe>
          {({ size }) => (
            <Document
              file={storyPrompt}
              onLoadSuccess={onDocumentLoadSuccess}
              loading="Loading Story..."
            >
              <div className="page-container">
                <Page
                  width={size.width ? size.width : 1}
                  pageNumber={pageNumber}
                />
                <p>
                  Page {pageNumber} of {numPages}
                </p>
              </div>
            </Document>
          )}
        </SizeMe>
        <div className="btn-container">
          <Button
            type="button"
            disabled={pageNumber <= 1}
            onClick={previousPage}
          >
            Previous Page
          </Button>
          <Button
            type="button"
            disabled={pageNumber >= numPages}
            onClick={nextPage}
          >
            Next Page
          </Button>
        </div>
      </div>
    </>
  );
};
export default RenderStoryViewer;
