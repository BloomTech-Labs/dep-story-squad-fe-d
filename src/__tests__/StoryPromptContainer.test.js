import * as React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import LoadingComponent from '../components/common/ParentLoadingComponent';
import RenderStoryViewer from '../components/pages/StoryPrompt/RenderStoryViewer';
import StoryPromptContainer from '../components/pages/StoryPrompt/StoryViewerContainer';
jest.mock('@okta/okta-react', () => ({
  useOktaAuth: () => {
    return {
      authState: {
        isAuthenticated: true,
      },
      authService: {},
    };
  },
}));

configure({ adapter: new Adapter() });

describe('<ProfileModalContainer />', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore([]);
  const store = mockStore();

  describe('Render <ProfileModalContainer />', () => {
    let shallowWrapper;
    beforeEach(() => {
      shallowWrapper = shallow(
        <Router>
          <Provider store={store}>
            <StoryPromptContainer />
          </Provider>
        </Router>
      ).dive();
    });

    it('Find RenderStoryViewer', () => {
      expect(shallowWrapper.find(RenderStoryViewer));
    });
    it('Find Loading Component', () => {
      expect(shallowWrapper.find(LoadingComponent));
    });
    it('Find StoryPromptContainer', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
