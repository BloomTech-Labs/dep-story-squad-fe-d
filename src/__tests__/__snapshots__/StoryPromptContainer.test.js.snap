// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileModalContainer /> Render <ProfileModalContainer /> Find StoryPromptContainer 1`] = `ShallowWrapper {}`;
